src/transformStateWithClones.js

```diff
@@ -6,8 +6,47 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const transformLogState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(newState, action.extraData);
+        transformLogState.push({ ...newState });
+        break;
+
+      case 'removeProperties':
+        removeProperties(newState, action.keysToRemove);
+        transformLogState.push({ ...newState });
```

Comment:
[DONT REPEAT YOURSELF] - This comment is repeated for line 23. You should push the state copy to the array outside of the switch block to avoid code duplication.